test/dotnet-cache.Tests/GivenDotnetCachesAndPublishesProjects.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.PlatformAbstractions;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.DotNet.Cli.Publish.Tests
+{
+    public class GivenDotnetPublishPublisheswithCacheProjects : TestBase
+    {
+        private static string _tfm = "netcoreapp2.0";
+        private static string _frameworkVersion = Microsoft.DotNet.TestFramework.TestAssetInstance.CurrentRuntimeFrameworkVersion;
+
+        [Fact]
+        public void ItPublishesARunnablePortableApp()
+        {
+            var testAppName = "NewtonSoftDependentProject";
+            var profileProjectName = "NewtonsoftFilterProfile";
+
+            var testInstance = TestAssets.Get(testAppName)
+                .CreateInstance()
+                .WithSourceFiles()
+                .UseCurrentRuntimeFrameworkVersion();
+
+            var testProjectDirectory = testInstance.Root.ToString();
+            var rid = DotnetLegacyRuntimeIdentifiers.InferLegacyRestoreRuntimeIdentifier();
+            var localAssemblyCache = Path.Combine(testProjectDirectory, "localAssemblyCache");
+            var intermediateWorkingDirectory = Path.Combine(testProjectDirectory, "workingDirectory");
+            var profileProject = Path.Combine(TestAssetsManager.AssetsRoot, profileProjectName, $"{profileProjectName}.xml");
+
+            new RestoreCommand()
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute()
+                .Should().Pass();
+
+            new CacheCommand()
+                .WithEntries(profileProject)
+                .WithFramework(_tfm)
+                .WithRuntime(rid)
+                .WithOutput(localAssemblyCache)
+                .WithRuntimeFrameworkVersion(_frameworkVersion)
+                .WithIntermediateWorkingDirectory(intermediateWorkingDirectory)
+                .Execute($"--preserve-working-dir")
```

Comment:
That is true. You don't need to check that, but if you are passing these parameters as some form of validation, you should probably assert something on it. Otherwise, like I said, you are just testing that parsing these strings won't throw.